docs/proposals/kubelet-systemd.md

```diff
@@ -0,0 +1,277 @@
+for non-Kubernetes components:
+
+* `--system-reserved=:` - A set of `ResourceName`=`ResourceQuantity` pairs that
+describe resources reserved for host daemons.
+
+The `kubelet` does not enforce `--system-reserved`, but the ability to distinguish
+the static reservation from observed usage is important for node accounting.
+
+The `kubelet` will accept a new flag in a `systemd` environment to support accounting
+system-reserved resources.
+
+* `--slice-system="system.slice"` - Slice reserved for
+system usage when using systemd cgroup driver.
+
+The `kubelet` will error in any of the following conditions:
+* the named `slice` does not exist
+* if the `cpu` and `memory` controllers are not enabled on the specified slice
+
+### Kubernetes reserved
+
+The node may set aside a set of resources for Kubernetes components: 
+
+* `--kube-reserved=:` - A set of `ResourceName`=`ResourceQuantity` pairs that
+describe resources reserved for host daemons.
+
+The `kubelet` does not enforce `--kube-reserved` at this time, but the ability
+to distinguish the static reservation from observed usage is important for node accounting.
+
+This proposal asserts that all components that Kubernetes brings to the node are
+parented by a common `slice` in a separate part of the hierarchy than system-reserved
+resources.
+
+This proposal asserts that `kubernetes.slice` is the default slice associated with
+the `kubelet` and `kube-proxy` service units defined in the project.
+
+The `kubelet` will detect the parent `slice` from which it was launched to track
+kubernetes-reserved observed usage.
+
+If the `kubelet` is parented by a `slice` that is categorized as system-reserved,
+the `kubelet` will log a warning that kubernetes-reserved observed usage accounting is
+disabled, but the static reservation is still observed for purposes of reporting the
+allocable resources for the node.
+
+If the `kubelet` is launched directly from a terminal, it's most likely destination will
+be in a `scope` that is a child of `user.slice` as follows:
+
+`/sys/fs/cgroup/<controller>/user.slice/user-1000.slice/session-1.scope`
+
+In this context, the parent `scope` is what will be used to facilitate local developer
+debugging scenarios for tracking `kube-reserved` usage.
+
+### Kubernetes container runtime reserved
+
+This proposal asserts that the reservation of compute resources for any associated
+container runtime daemons is tracked by the operator under the system-reserved or
+kubernetes-reserved values at this time and any enforced limits are set by the
+operator specific to the container runtime.
+
+For future consideration, if the `kubelet` attempts to enforce `--kube-reserved`, this
+proposal asserts that the reservation associated with any particular container runtime
+should be split out at that time.  In a `systemd` environment, this proposal asserts that
+you cannot mandate the container runtime daemon is in the same `slice` as the Kubernetes
```

Comment:
If a kube node is completely owned by kubelet, why not let kubelet do what it has to do? Are we trying to make a generic Linux distribution also work very well as a kube node?